File: 1920-team1-backend/WebApplication/WebApplication/Models/Internship.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using System.Runtime.Serialization;
using WebApplication.Enums;

namespace WebApplication.Models
{
    [DataContract]
    [Serializable]
    public class Internship : BaseModel
    {
        public Internship()
        {
            Courses = new List<InternshipCourse>();
            Technologies = new List<InternshipTechnology>();
            Validations = new List<Validation>();
            Students = new List<InternshipStudent>();
            Promotors = new List<InternshipPerson>();
            Periods = new List<InternshipPeriod>();
        }
        [DataMember]
        [Required]
        [NotNull]
        public Company Company { get; set; }
        
        [DataMember]
        [Required]
        [NotNull]
        public string Title { get; set; }
        
        [DataMember]
        [Required]
        [NotNull]
        public string Description { get; set; }

        [DataMember]
        [Required]
        [NotNull]
        public ICollection<InternshipCourse> Courses { get; }

        [DataMember]
        [Required]
        [NotNull]
        public ICollection<InternshipTechnology> Technologies { get; }

        [DataMember]
        [Required]
        [NotNull]
        public Person ContactPerson { get; set; }
                
        [DataMember]
        [Required]
        [NotNull]
        public ICollection<InternshipPerson> Promotors { get; }

        [DataMember]
        [Required]
        public Address Address { get; set; }
        
        [DataMember]
        [Required]
        [NotNull]
        public string TechDescription { get; set; }
        
        [DataMember]
        [Required]
        [NotNull]
        public string ResearchTopic { get; set; }
        
        [DataMember]
        [Required]
        public bool? Application { get; set; }
        
        [DataMember]
        [Required]
        public bool? Résumée { get; set; }
        
        [DataMember]
        [Required]
        public bool? Reimbursement { get; set; }
        
        [DataMember]
        [Range(0,2)]
        public int StudentAmount { get; set; }
        
        [DataMember]
        [Required]
        [NotNull]
        public ICollection<InternshipStudent> Students { get; }

        [DataMember]
        [Required]
        [NotNull]
        public int NrOfSupportEmployees { get; set; }

        [DataMember]
        public string Remarks { get; set; }

        [DataMember]
        public string Conditions { get; set; }

        [DataMember]
        [Required]
        [NotNull]
        public ICollection<Validation> Validations { get; }
        
        [DataMember]
        [Required]
        [NotNull]
        public ICollection<InternshipPeriod> Periods { get; }

        [DataMember]
        [Required]
        [NotNull]
        public InternshipStateEnum State { get; set; }
        
        [DataMember]
        public FeedBack FeedBack { get; set; }
    }
}

File: 1920-team1-frontend/ITProjectTeam1/src/app/models/internship.ts
import {Course} from './course';
import {Technology} from './technology';
import {Person} from './person';
import {Address} from './address';
import {Student} from './student';
import {Validation} from './validation';
import {Period} from './period';
import {InternshipState} from '../enums/internship-state.enum';
import {Feedback} from './feedback';

export interface Internship {
  // TO BE CHECKED
  id?: number;
  courses?: Course[];
  technologies?: Technology[];
  students?: Student[];
  validations?: Validation[];
  periods?: Period[];
  state?: InternshipState;


  technologyIds: number[];
  courseIds: number[];
  periodIds: number[];
  studentIds: number[];
  title: string;
  description: string;
  contactPerson: Person;
  promotors: Person[];
  address: Address;
  techDescription: string;
  researchTopic: string;
  application: boolean;
  resumee: boolean;
  reimbursement: boolean;
  studentAmount: number;
  remarks: string;
  nrOfSupportEmployees: number;
  conditions?: string;
  companyId?: string;
  favorite?: boolean;
  feedBack?: Feedback;
}
